go/test/endtoend/vreplication/fk_ext_test.go

```diff
@@ -0,0 +1,444 @@
+	cellName := fkextConfig.cell
+	cell := vc.Cells[fkextConfig.cell]
+	vc.AddKeyspace(t, []*Cell{cell}, keyspaceName, shards, vschema, schema, numReplicas, 0, tabletId, nil)
+	for i, shard := range strings.Split(shards, ",") {
+		require.NoError(t, vtgate.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.replica", keyspaceName, shard), 1, shardStatusWaitTimeout))
+		tablets[shard] = vc.Cells[cellName].Keyspaces[keyspaceName].Shards[shard].Tablets[fmt.Sprintf("%s-%d", cellName, tabletId+i*100)].Vttablet
+	}
+	err := vc.VtctldClient.ExecuteCommand("RebuildVSchemaGraph")
+	require.NoError(t, err)
+	require.NoError(t, waitForColumn(t, vtgate, keyspaceName, "parent", "id"))
+	require.NoError(t, waitForColumn(t, vtgate, keyspaceName, "child", "parent_id"))
+	return tablets
+}
+
+func doMoveTables(t *testing.T, sourceKeyspace, targetKeyspace, workflowName, tabletTypes string, targetTabs map[string]*cluster.VttabletProcess, atomicCopy bool) {
+	mt := newMoveTables(vc, &moveTablesWorkflow{
+		workflowInfo: &workflowInfo{
+			vc:             vc,
+			workflowName:   workflowName,
+			targetKeyspace: targetKeyspace,
+			tabletTypes:    tabletTypes,
+		},
+		sourceKeyspace: sourceKeyspace,
+		atomicCopy:     atomicCopy,
+	}, workflowFlavorRandom)
+	mt.Create()
+
+	waitForWorkflowState(t, vc, fmt.Sprintf("%s.%s", targetKeyspace, workflowName), binlogdatapb.VReplicationWorkflowState_Running.String())
+
+	for _, targetTab := range targetTabs {
+		catchup(t, targetTab, workflowName, "MoveTables")
+	}
+	vdiff(t, targetKeyspace, workflowName, fkextConfig.cell, false, true, nil)
+	lg.Stop()
+	lg.SetDBStrategy("vtgate", targetKeyspace)
+	if lg.Start() != nil {
+		t.Fatal("Start failed")
+	}
+
+	mt.SwitchReadsAndWrites()
+
+	if lg.WaitForAdditionalRows(100) != nil {
+		t.Fatal("WaitForAdditionalRows failed")
+	}
+
+	waitForLowLag(t, sourceKeyspace, workflowName+"_reverse")
+	vdiff(t, sourceKeyspace, workflowName+"_reverse", fkextConfig.cell, false, true, nil)
+	if lg.WaitForAdditionalRows(100) != nil {
+		t.Fatal("WaitForAdditionalRows failed")
+	}
+
+	mt.ReverseReadsAndWrites()
+	if lg.WaitForAdditionalRows(100) != nil {
+		t.Fatal("WaitForAdditionalRows failed")
+	}
+	waitForLowLag(t, targetKeyspace, workflowName)
+	time.Sleep(5 * time.Second)
+	vdiff(t, targetKeyspace, workflowName, fkextConfig.cell, false, true, nil)
+	lg.Stop()
+	mt.SwitchReadsAndWrites()
+	mt.Complete()
+	if err := vc.VtctldClient.ExecuteCommand("ApplyRoutingRules", "--rules={}"); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func importIntoVitess(t *testing.T) {
+	targetTabs := newKeyspace(t, fkextConfig.target1KeyspaceName, "0", FKExtTarget1VSchema, FKExtSourceSchema, 200, 1)
+	doMoveTables(t, fkextConfig.sourceKeyspaceName, fkextConfig.target1KeyspaceName, "import", "primary", targetTabs, true)
+}
+
+const getConstraintsQuery = `
+SELECT CONSTRAINT_NAME, TABLE_NAME 
+FROM INFORMATION_SCHEMA.KEY_COLUMN_USAGE 
+WHERE TABLE_SCHEMA = ` + "'%s'" + ` AND REFERENCED_TABLE_NAME IS NOT NULL;
```

Comment:
Not sure I see the need to break the string.
```suggestion
WHERE TABLE_SCHEMA = `'%s'` AND REFERENCED_TABLE_NAME IS NOT NULL;
```

Reply:
Done